### PR TITLE
New Rule - Toggle Bool

### PR DESCRIPTION
**Pull request by daltonclaybrook**
This PR adds a new rule for preferring `Bool.toggle()` over negation in Swift 4.2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,11 @@
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#2123](https://github.com/realm/SwiftLint/issues/2123)
 
+* Add `toggle_bool` opt-in rule which suggests using `Bool.toggle()` over 
+  `someBool = !someBool`. Requires Swift 4.2.
```
Comment (ornithocoder) [inline CHANGELOG.md]: Please make sure there are two trailing spaces on the last line describing the change. See [CONTRIBUTING.md](https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md#tracking-changes)

Comment (daltonclaybrook) [reply CHANGELOG.md]: Whoops, must have missed that one. 😅

File: Source/SwiftLintFramework/Rules/Idiomatic/ToggleBoolRule.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+public struct ToggleBoolRule: ConfigurationProviderRule, OptInRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static var description = RuleDescription(
+        identifier: "toggle_bool",
+        name: "Toggle Bool",
+        description: "Prefer `Bool.toggle()` over `someBool = !someBool`.",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Idiomatic/ToggleBoolRule.swift]: should we use `someBool.toggle()` instead of `Bool.toggle()` to be more explicit?

Comment (daltonclaybrook) [reply Source/SwiftLintFramework/Rules/Idiomatic/ToggleBoolRule.swift]: Yah, I think that makes more sense. I've also updated the Changelog to match.

**Pull request merged**